gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,588 @@
+      Map<String, Object> args = new HashMap<>();
+      Map<String, Object> nestedArgs = new HashMap<>();
+      nestedArgs.put("domain", domain);
+      nestedArgs.put("message", message);
+      nestedArgs.put("reason", reason);
+      List<Map<String, Object>> errors = ImmutableList.of(nestedArgs);
+      args.put("errors", errors);
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      try {
+        return jsonFactory.toString(ImmutableMap.of("error", args));
+      } catch (IOException e) {
+        throw new RuntimeException("Error when generating JSON error response.");
+      }
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      // see https://cloud.google.com/resource-manager/reference/rest/
+      String path = exchange.getRequestURI().getPath();
+      String requestMethod = exchange.getRequestMethod();
+      Response response = null;
+      if (requestMethod.equals("POST") && path.startsWith("/v1beta1/projects")) {
+        if (path.contains("undelete")) {
+          response = undelete(projectIdFromURI(path));
+        } else {
+          String requestBody =
+              decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+          response = create(jsonFactory.fromString(requestBody, Project.class));
+        }
+      } else if (requestMethod.equals("DELETE")) {
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET") && path.startsWith("/v1beta1/projects/")) {
+        response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+      } else if (requestMethod.equals("GET")) {
+        response = list(parseListOptions(exchange.getRequestURI().getQuery()));
+      } else if (requestMethod.equals("PUT")) {
+        String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+        response = replace(jsonFactory.fromString(requestBody, Project.class));
+      }
+      if (response == null) {
+        throw new UnsupportedOperationException("Request not recognized.");
+      }
+      exchange.getResponseHeaders().set("Content-type", "application/json; charset=UTF-8");
+      exchange.sendResponseHeaders(response.code(), response.body().length());
+      OutputStream outputStream = exchange.getResponseBody();
+      outputStream.write(response.body().getBytes());
+      outputStream.close();
+    }
+  }
+
+  private static String decodeContent(Headers headers, InputStream inputStream) throws IOException {
+    List<String> contentEncoding = headers.get("Content-encoding");
+    byte[] bytes;
+    if (contentEncoding != null && contentEncoding.size() > 0
+        && contentEncoding.get(0).contains("gzip")) {
+      bytes = ByteStreams.toByteArray(new GZIPInputStream(inputStream));
```

Comment:
`input = new GZIPInputStream(inputStream)` and then you can share the code of toByteArray.